File: Java/Sostituzione.java

public class Sostituzione extends Frase implements X{
	private char[] Risultato = new char[26]; //Testo cifrato
	private int N; //Numero di spostamenti
	
	/* Controllo se n e' corretto */
	public Sostituzione(String phrase, int n) throws Errori{
		super(phrase);
		if(n < 0 || n > 26) {
			n = -1;
			throw new Errori("Numero troppo grande");
		}
		N = n;
	}
	/* Ritorno il testo cifrato */
	public char[] getRisultato() {
		return Risultato;
	}
	/* Cifro il testo in chiaro */
	public void codifica() {
        String f = getPhrase();
        for(int i=0;i<f.length();++i){
            if(f.charAt(i) == ' ') Risultato[i] = ' ';
            else Risultato[i] = getLettera(f.charAt(i),N);
        }
    }
	/* Stampo il testo cifrato */
	public void stampa() {
        System.out.println("Frase risultante: ");
        for(int i=0;i<getPhrase().length();++i) System.out.print(Risultato[i]);
    }
}

File: C++/sostituzione.cpp
#include "sostituzione.h"
#include <iostream>
/* Controllo se 'n' è corretto */
    Sostituzione::Sostituzione(int n,string p):Frase(p),N(n){
        if(n<0 || n>26) Frase::warning("Il numero di spostamenti deve essere dentro il range 0-26!");
    }
/* Costruisco il testo cifrato */
    void Sostituzione::codifica() {
        string f = getPhrase();
        for(int i=0;i<f.size();i++){
            if(f.at(i) == ' ') Risultato[i] = ' ';
            else Risultato[i] = getLettera(f.at(i),N);
        }
    }
/* Stampa */
    void Sostituzione::stampa() const{
        cout<<"Frase codificata usando la sostituzione: ";
        for(int i=0;i<getPhrase().size();i++) cout<<Risultato[i];
        cout<<endl;
    }
/* Ritorno il testo cifrato */
    char* Sostituzione::getRisultato() const{
        char* p = new char[getPhrase().size()];
        strcpy(p,Risultato);
        return p;
    }
